### PR TITLE
F1266 env var interpolation

### PR DESCRIPTION
**Pull request by ssickles**
Added the ns_env_variables data source and incorporated to support env_variables and secrets.

### CODE REVIEW

File: urls.tf [hunk 1]
```diff
@@ -0,0 +1,35 @@
+locals {
+  // Private and public URLs are shown in the Nullstone UI
+  // Typically, they are created through capabilities attached to the application
+  // If this module has URLs, add them here as list(string)
+  additional_private_urls = []
```
Comment (BSick7) [inline urls.tf:5]: These are defined in `outputs.tf`.

File: urls.tf [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  additional_public_urls = []
+
+  private_urls = concat([for url in try(local.capabilities.private_urls, []) : url["url"]], local.additional_private_urls)
```
Comment (BSick7) [inline urls.tf:8]: Make sure to update `outputs.tf` to use these.

File: env_vars.tf [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    NULLSTONE_STACK         = data.ns_workspace.this.stack_name
+    NULLSTONE_APP           = data.ns_workspace.this.block_name
+    NULLSTONE_ENV           = data.ns_workspace.this.env_name
+    NULLSTONE_VERSION       = data.ns_app_env.this.version
+    NULLSTONE_COMMIT_SHA    = data.ns_app_env.this.commit_sha
+    NULLSTONE_PUBLIC_HOSTS  = join(",", local.public_hosts)
+    NULLSTONE_PRIVATE_HOSTS = join(",", local.private_hosts)
+  })
+
+  input_env_vars = merge(local.standard_env_vars, local.cap_env_vars, local.app_secret_ids, var.service_env_vars)
```
Comment (BSick7) [inline env_vars.tf:32]: `app_secrets_ids` aren't interpolated, but I'm not sure if that's critical right now.
We don't show these in the UI today.

File: urls.tf (hunk 1 above)
Comment (ssickles) [reply urls.tf:5]: I removed these from outputs.tf since they are used here.

File: env_vars.tf (hunk 3 above)
Comment (ssickles) [reply env_vars.tf:32]: I changed this to not send in the app_secret_ids to the ns_env_variables data source anymore. Instead I am merging those ids into the `all_env_vars` after interpolation is finished.

**Review by BSick7 — APPROVED** [review]
Looks good!

**Pull request merged**